src/rendering/vcAtmosphereRenderer.cpp

```diff
@@ -0,0 +1,323 @@
+#include "vcAtmosphereRenderer.h"
+#include "vcState.h"
+#include "vcInternalModels.h"
+
+#include "gl/vcShader.h"
+
+#include "udGeoZone.h"
+#include "udMath.h"
+#include "udPlatform.h"
+#include "udStringUtil.h"
+
+#include "gl/vcMesh.h"
+
+#include "atmosphere/model.h"
+
+enum Luminance
+{
+  // Render the spectral radiance at kLambdaR, kLambdaG, kLambdaB.
+  NONE,
+  // Render the sRGB luminance, using an approximate (on the fly) conversion
+  // from 3 spectral radiance values only (see section 14.3 in <a href=
+  // "https://arxiv.org/pdf/1612.04336.pdf">A Qualitative and Quantitative
+  //  Evaluation of 8 Clear Sky Models</a>).
+  APPROXIMATE,
+  // Render the sRGB luminance, precomputed from 15 spectral radiance values
+  // (see section 4.4 in <a href=
+  // "http://www.oskee.wz.cz/stranka/uploads/SCCG10ElekKmoch.pdf">Real-time
+  //  Spectral Scattering in Large-scale Natural Participating Media</a>).
+  PRECOMPUTED
+};
+
+struct vcAtmosphereRenderer
+{
+  atmosphere::Model *pModel;
+
+  // some model params
+  Luminance use_luminance;
+  double sun_zenith_angle_radians;
+  double sun_azimuth_angle_radians;
+  double exposure;
+
+  struct
+  {
+    vcShader *pProgram;
+    vcShaderSampler *uniform_transmittance;
+    vcShaderSampler *uniform_scattering;
+    vcShaderSampler *uniform_irradiance;
+    vcShaderSampler *uniform_sceneColour;
+    vcShaderSampler *uniform_sceneDepth;
+    vcShaderConstantBuffer *uniform_vertParams;
+    vcShaderConstantBuffer *uniform_fragParams;
```

Comment:
None of these are named correctly.

Reply:
They aren't? They're consistent with other uniform struct variables.

Reply:
Are they? I hadn't noticed we had a different set of rules for these. I would have expected `pUniformFragParams` for example, so `p` prefix and no underscores.

Reply:
Agreed. There are a few things like this in our renderer the need to be cleaned up.